.github/actions/k-bench/evaluate/compare.py

```diff
@@ -0,0 +1,114 @@
+"""Compare the current benchmark data against the previous."""
+import os
+import json
+from typing import Tuple
+
+# Progress indicator icons
+PROGRESS = ['⬇️', '⬆️']
+
+# List of benchmarks for which higher numbers are better
+BIGGER_BETTER = [
+    'net_internode_snd',
+    'net_intranode_snd',
+    'fio_root_async_R70W30_R',
+    'fio_root_async_R70W30_W',
+    'fio_root_async_R100W0_R',
+    'fio_root_async_R0W100_W',
+]
+
+# Lookup for test suite -> unit
+UNIT_STR = {
+    'net_internode_snd': 'Mbit/s',
+    'net_intranode_snd': 'Mbit/s',
+    'fio_root_async_R70W30_R': 'MiB/s',
+    'fio_root_async_R70W30_W': 'MiB/s',
+    'fio_root_async_R100W0_R': 'MiB/s',
+    'fio_root_async_R0W100_W': 'MiB/s',
+}
+# API units are ms, so this is shorter than cluttering the dictionary:
+API_UNIT_STR = "ms"
+
+
+def is_bigger_better(bench_suite: str) -> bool:
+    return bench_suite in BIGGER_BETTER
+
+
+def get_paths() -> Tuple[str, str]:
+    """Read the benchmark data paths.
+
+    Expects ENV vars (required):
+    - PREV_BENCH=/path/to/previous.json
+    - CURR_BENCH=/path/to/current.json
+
+    Raises TypeError if at least one of them is missing.
+
+    Returns: a tuple of (prev_bench_path, curr_bench_path).
+    """
+    path_prev = os.environ.get('PREV_BENCH', None)
+    path_curr = os.environ.get('CURR_BENCH', None)
+    if not path_prev or not path_curr:
+        raise TypeError(
+            'Both ENV variables PREV_BENCH and CURR_BENCH are required.')
+    return path_prev, path_curr
+
+
+def main() -> None:
+    """Compare the current benchmark data against the previous.
+
+    Create a markdown table showing the benchmark progressions.
+
+    Print the result to stdout.
+    """
+    path_prev, path_curr = get_paths()
+
+    with open(path_prev) as f_prev:
+        bench_prev = json.load(f_prev)
+    with open(path_curr) as f_curr:
+        bench_curr = json.load(f_curr)
+
+    name = bench_curr['subject']
+    if name != bench_prev['subject']:
```

Comment:
Presence of these keys is not checked. You could use `in` or `try`..`catch KeyError`.